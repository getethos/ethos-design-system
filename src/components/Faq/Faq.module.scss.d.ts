// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'body': string;
  'carrot': string;
  'leftColumn': string;
  'phoneTitle': string;
  'rightColumn': string;
  'root': string;
  'summary': string;
}
export const cssExports: CssExports;
export default cssExports;
