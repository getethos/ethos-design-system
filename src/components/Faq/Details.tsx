import * as React from 'react'
import { Body } from '../index'
import styles from './Faq.module.scss'
type DetailsProps = {
  summary: string
  open?: boolean
}
export const Details: React.SFC<DetailsProps> = ({
  children,
  summary,
  open = false,
}) => {
  const carrotSvg = (
    <svg viewBox="0 0 19 18" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M0.999999 8.375L9.375 16.75L17.75 8.375"
        stroke="black"
        strokeOpacity="0.85"
        strokeWidth="1.5"
      />
    </svg>
  )
  return (
    <details open={open}>
      <summary>
        <div className={styles.summary}>
          <Body.Medium500>{summary}</Body.Medium500>
          <div className={styles.carrot}>{carrotSvg}</div>
        </div>
      </summary>
      <div className={styles.body}>
        <Body.Regular400>{children}</Body.Regular400>
      </div>
    </details>
  )
}
