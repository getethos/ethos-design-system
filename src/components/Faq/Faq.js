import React from 'react'
import PropTypes from 'prop-types'
import uuidv4 from 'uuid/v4'
import { TitleLarge, TitleMedium, Body } from '../index'

import symbol from '../../svgs/expand-details.svg'
import styles from './Faq.module.scss'

export const Faq = ({ questions, open }) => {
  return (
    <div className={styles.root}>
      <div className={styles.leftColumn}>
        <TitleMedium.Serif.Book500>FAQs</TitleMedium.Serif.Book500>
      </div>

      <div className={styles.rightColumn}>
        <div className={styles.phoneTitle}>
          <TitleLarge.Serif.Book500>FAQs</TitleLarge.Serif.Book500>
        </div>
        {questions.map((q, index) => {
          const startOpen = open && index === 0
          return (
            <Details summary={q.summary} open={startOpen} key={uuidv4()}>
              {q.content}
            </Details>
          )
        })}
      </div>
    </div>
  )
}

Faq.defaultProps = {
  questions: [],
  open: true,
}

Faq.propTypes = {
  questions: PropTypes.arrayOf(
    PropTypes.shape({
      summary: PropTypes.string,
      content: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    })
  ),
  /** Set to false for all questions to start collapsed. */
  open: PropTypes.bool,
}

const Details = ({ children, summary, open }) => {
  const carrotSvg = (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox={symbol.viewBox}
      className={styles.carrotSvg}
    >
      <use xlinkHref={`#${symbol.id}`} />
    </svg>
  )

  return (
    <details open={open}>
      <summary>
        <div className={styles.summary}>
          <Body.Medium500>{summary}</Body.Medium500>
          <div className={styles.carrot}>{carrotSvg}</div>
        </div>
      </summary>
      <div className={styles.body}>
        <Body.Regular400>{children}</Body.Regular400>
      </div>
    </details>
  )
}

Details.defaultProps = {
  open: false,
}

Details.propTypes = {
  children: PropTypes.node.isRequired,
  summary: PropTypes.string.isRequired,
  open: PropTypes.bool,
}

export default Faq
