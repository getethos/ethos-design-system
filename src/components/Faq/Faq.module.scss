@import '../Media/Media.scss';

.root {
  display: flex;

  details {
    display: block;
    width: 100%;
    margin-bottom: var(--Space-40);

    .body {
      color: var(--GraySecondary--opaque);
    }
  }
  .summary {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: var(--Space-8);
    outline: none;
    cursor: pointer;
    font-weight: bold;

    svg {
      width: 20px;
      height: 20px;
      transition: 0.25s ease-in-out;
      transform: rotate(0deg);
    }
  }
  summary {
    outline: none;
    list-style: none;
  }

  details[open] {
    svg {
      transition: 0.25s ease-in-out;
      transform: rotate(-180deg);
    }
  }
}

.leftColumn {
  width: 50%;
  @include for-phone-only {
    display: none;
  }
}

.rightColumn {
  display: flex;
  flex-direction: column;
  width: 50%;
  @include for-phone-only {
    width: 100%;
  }

  @include for-tablet-and-up {
    margin-top: 7px;
  }
}

.phoneTitle {
  @include for-phone-only {
    visibility: visible;
    margin-bottom: var(--Space-48);
  }
  @include for-tablet-and-up {
    visibility: hidden;
    height: 0px;
  }
}

.carrot {
  margin-left: var(--Space-24);
}
