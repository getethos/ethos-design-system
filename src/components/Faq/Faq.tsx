import * as React from 'react'
import { v4 as uuidv4 } from 'uuid'
import { TitleLarge, TitleMedium } from '../index'
import { Details } from './Details'
import styles from './Faq.module.scss'
type FaqProps = {
  questions?: {
    summary: string
    content: string | JSX.Element
  }[]
  open?: boolean
}
export const Faq: React.SFC<FaqProps> = ({ questions = [], open = true }) => {
  return (
    <div className={styles.root}>
      <div className={styles.leftColumn}>
        <TitleMedium.Serif.Book500>FAQs</TitleMedium.Serif.Book500>
      </div>

      <div className={styles.rightColumn}>
        <div className={styles.phoneTitle}>
          <TitleLarge.Serif.Book500>FAQs</TitleLarge.Serif.Book500>
        </div>
        {questions.map(({ summary, content }, index) => {
          const startOpen = open && index === 0
          return (
            <Details summary={summary} open={startOpen} key={uuidv4()}>
              {content}
            </Details>
          )
        })}
      </div>
    </div>
  )
}
export default Faq
