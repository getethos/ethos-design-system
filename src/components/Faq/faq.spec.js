import React from 'react'
import { Faq } from './Faq.js'
import renderer from 'react-test-renderer'

describe('Faq', () => {
  describe('API', () => {
    test('exports properly', () => {
      expect(Faq).toBeDefined()
    })
  })

  describe('rendering component', () => {
    test('default', () => {
      const tree = renderer.create(<Faq />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    test('default questions', () => {
      const demoQuestions = [
        {
          summary: 'First',
          content: (
            <div>
              <p>
                Culpa occaecat nulla tempor in veniam exercitation qui in
                voluptate ullamco adipisicing fugiat. Cupidatat do officia est
                Lorem anim sunt enim aliqua deserunt ut aliqua excepteur. Et
                aliqua elit in dolor reprehenderit cillum ipsum irure aliquip
                duis labore ipsum.
              </p>
              <p>
                Excepteur et laborum ea enim dolor sit Lorem et cupidatat veniam
                pariatur consequat. Occaecat amet aliqua sunt exercitation ut
                quis consectetur anim aliquip ex non laboris. Id proident labore
                et ex laborum.
              </p>
            </div>
          ),
        },
        {
          summary: 'Second',
          content:
            'Qui ad sit sunt dolore reprehenderit irure ad. Elit ipsum elit minim labore cupidatat duis est ad incididunt ea. Ea ut aliqua et reprehenderit Lorem commodo reprehenderit et commodo voluptate do proident. Fugiat amet elit commodo ad officia est irure mollit elit. Ullamco voluptate in elit commodo nulla. Labore anim do laborum laborum aliquip est cillum qui magna enim. Et nisi non laborum magna ullamco deserunt est duis cillum consectetur.',
        },
        {
          summary: 'Third',
          content:
            'Ea adipisicing excepteur nulla veniam mollit quis laborum anim quis et nisi officia. Ad eu pariatur duis duis dolore duis commodo non. Aute ipsum ex est cillum ut. Magna quis eu tempor do ullamco. Fugiat voluptate incididunt deserunt consectetur Lorem esse sit et cillum nisi.',
        },
      ]
      const tree = renderer.create(<Faq questions={demoQuestions} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
