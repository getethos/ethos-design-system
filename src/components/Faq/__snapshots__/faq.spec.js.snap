// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faq rendering component default 1`] = `
<div
  className="root"
>
  <div
    className="leftColumn"
  >
    <div
      className="TitleMedium Cambon Book500"
    >
      FAQs
    </div>
  </div>
  <div
    className="rightColumn"
  >
    <div
      className="phoneTitle"
    >
      <div
        className="TitleLarge Cambon Book500"
      >
        FAQs
      </div>
    </div>
  </div>
</div>
`;

exports[`Faq rendering component default questions 1`] = `
<div
  className="root"
>
  <div
    className="leftColumn"
  >
    <div
      className="TitleMedium Cambon Book500"
    >
      FAQs
    </div>
  </div>
  <div
    className="rightColumn"
  >
    <div
      className="phoneTitle"
    >
      <div
        className="TitleLarge Cambon Book500"
      >
        FAQs
      </div>
    </div>
    <details
      open={true}
    >
      <summary>
        <div
          className="summary"
        >
          <div
            className="Body Theinhardt Medium500"
          >
            First
          </div>
          <div
            className="carrot"
          >
            <svg
              className="carrotSvg"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlinkHref="#undefined"
              />
            </svg>
          </div>
        </div>
      </summary>
      <div
        className="body"
      >
        <div
          className="Body Theinhardt Regular400"
        >
          <div>
            <p>
              Culpa occaecat nulla tempor in veniam exercitation qui in voluptate ullamco adipisicing fugiat. Cupidatat do officia est Lorem anim sunt enim aliqua deserunt ut aliqua excepteur. Et aliqua elit in dolor reprehenderit cillum ipsum irure aliquip duis labore ipsum.
            </p>
            <p>
              Excepteur et laborum ea enim dolor sit Lorem et cupidatat veniam pariatur consequat. Occaecat amet aliqua sunt exercitation ut quis consectetur anim aliquip ex non laboris. Id proident labore et ex laborum.
            </p>
          </div>
        </div>
      </div>
    </details>
    <details
      open={false}
    >
      <summary>
        <div
          className="summary"
        >
          <div
            className="Body Theinhardt Medium500"
          >
            Second
          </div>
          <div
            className="carrot"
          >
            <svg
              className="carrotSvg"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlinkHref="#undefined"
              />
            </svg>
          </div>
        </div>
      </summary>
      <div
        className="body"
      >
        <div
          className="Body Theinhardt Regular400"
        >
          Qui ad sit sunt dolore reprehenderit irure ad. Elit ipsum elit minim labore cupidatat duis est ad incididunt ea. Ea ut aliqua et reprehenderit Lorem commodo reprehenderit et commodo voluptate do proident. Fugiat amet elit commodo ad officia est irure mollit elit. Ullamco voluptate in elit commodo nulla. Labore anim do laborum laborum aliquip est cillum qui magna enim. Et nisi non laborum magna ullamco deserunt est duis cillum consectetur.
        </div>
      </div>
    </details>
    <details
      open={false}
    >
      <summary>
        <div
          className="summary"
        >
          <div
            className="Body Theinhardt Medium500"
          >
            Third
          </div>
          <div
            className="carrot"
          >
            <svg
              className="carrotSvg"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlinkHref="#undefined"
              />
            </svg>
          </div>
        </div>
      </summary>
      <div
        className="body"
      >
        <div
          className="Body Theinhardt Regular400"
        >
          Ea adipisicing excepteur nulla veniam mollit quis laborum anim quis et nisi officia. Ad eu pariatur duis duis dolore duis commodo non. Aute ipsum ex est cillum ut. Magna quis eu tempor do ullamco. Fugiat voluptate incididunt deserunt consectetur Lorem esse sit et cillum nisi.
        </div>
      </div>
    </details>
  </div>
</div>
`;
